### PR TITLE
Communities 

### PR DESCRIPTION
**Pull request by cammellos**
This commit add the basics for community management.

- Create a community
- Import/Export a community
- Send an invitation message
- Add/Remove users to a community
- Create channels within a community

I have left the commit split in order to facilitate the review process.

### CODE REVIEW

**Comment by cammellos** [comment]
@Samyoul @jakubgs This should be ready for review. 
I will merge after Profile pics are merged, but it can already be reviewed, mind I have pulled some of the protobufs from there and will be replaced once profile pics is merged.
Thanks!

File: VERSION
Comment (Samyoul) [inline VERSION]: I feel like Communities should be a semver `minor` level, rather than a `patch` level change

File: protocol/chat.go
Comment (Samyoul) [inline protocol/chat.go:340]: I really like the neatness of this function

File: protocol/messenger_response.go
Comment (Samyoul) [inline protocol/messenger_response.go:26]: Ah, I like this,

File: services/ext/api.go
Comment (Samyoul) [inline services/ext/api.go]: I normally don't mind Exported functions not having a description comment, but on API endpoints, descriptions serve as a self documenting API. Do you think it would be beneficial to add description comments to these functions?

**Review by Samyoul — COMMENTED** [review]
So I've reviewed all the peripheral code and have no comments of substance. Looks good. I've reviewed 32 / 45 files, the remaining is the actually core of `protocol/communities`.

Question. What is a "grant"? It pops up as a new field in the protobufs, but I don't understand its role (which may be more obvious after I read the `protocol/communities code`)

**Comment by cammellos** [comment]
@Samyoul addressed feedback thanks!

Grant is described here: https://github.com/status-im/specs/pull/151/files

Basically because we can't rely on receiving messages in order, we have a `grant` that is given by the community creator to the invited user, which will be carried on when posting.
In this way other clients can optimistically accept the message if a grant is valid, in case the updated community description has not been fetched yet.
The feature is not fully implemented yet, but there's room for it in the codebase.

**Comment by Samyoul** [comment]
Ah thank you. So a grant is like an attestation or permission in the form of a signed message from a known admin of an organisation, that's a nice solution.

File: protocol/protobuf/chat_identity.proto
Comment (Samyoul) [inline protocol/protobuf/chat_identity.proto:23]: I like these as although they are noted as being organisations only, they can be used in a user context.

File: protocol/communities/community.go
Comment (Samyoul) [inline protocol/communities/community.go]: Do these functions need a `TODO` adding?

File: protocol/communities/community.go
Comment (Samyoul) [inline protocol/communities/community.go]: `signer` isn't used in this function

File: protocol/communities/community.go
Comment (Samyoul) [inline protocol/communities/community.go]: `chat` is explicitly declared and then immediately implicitly redeclared. I know that this is the only way to declare `ok` but as there is no code that relies on `chat` being conditionally set it is not necessary to explicitly declare it.

File: protocol/communities/community.go
Comment (Samyoul) [reply protocol/communities/community.go]: Ah, is this a work in progress?

File: protocol/communities/community.go
Comment (Samyoul) [inline protocol/communities/community.go]: `signer` not used in this function. But the function also looks like a work in progress, do we need to add a `TODO`?

File: protocol/communities/manager.go
Comment (Samyoul) [inline protocol/communities/manager.go:226]: `signer` and `invitation` are not used here. Is this a work in progress?

File: protocol/communities/manager_test.go
Comment (Samyoul) [inline protocol/communities/manager_test.go:61]: There seems to be a lot of functionality that `Manager` has that isn't tested here. Things like adding importing and exporting a community and handling description messages.

File: protocol/communities/persistence.go
Comment (Samyoul) [inline protocol/communities/persistence.go]: A tiny little thing, `rows.Close()` returns an error, so if anything goes wrong with closing the `rows` the error will never be handled.

If we care about this, something like below would work to catch the error.

```suggestion
func (p *Persistence) queryCommunities(query string) (response []*Community, err error) {
	rows, err := p.db.Query(query)
	if err != nil {
		return nil, err
	}
	defer func() {
		err = rows.Close()
	}()

```

File: protocol/communities_messenger_test.go
Comment (Samyoul) [inline protocol/communities_messenger_test.go]: `shh` is not used in this function, I can see that this suite manages shh so probably a hangover from other test suites that don't manage the protocol type.

File: protocol/message_handler.go
Comment (Samyoul) [inline protocol/message_handler.go]: `state` is not used in this function

**Review by Samyoul — APPROVED** [review]
Great work. Congratulations. My feedback is nothing major, mostly related to either unused parameters or how we handle certain erros.

**Pull request merged**
